examples/hosts/app-integration/external-data/src/model/containerCode.ts

```diff
@@ -23,34 +21,31 @@ const SignalType = {
 	ExternalDataChanged: "ExternalDataChange",
 };
 
+const taskListCollectionAlias = "task-list-collection";
 /**
  * {@inheritDoc ModelContainerRuntimeFactory}
  */
-export class TaskListContainerRuntimeFactory extends ModelContainerRuntimeFactory<IAppModel> {
+export class TaskListCollectionContainerRuntimeFactory extends ModelContainerRuntimeFactory<IAppModel> {
 	public constructor() {
 		super(
-			new Map([TaskListInstantiationFactory.registryEntry]), // registryEntries
+			new Map([TaskListCollectionInstantiationFactory.registryEntry]), // registryEntries
 		);
 	}
 
 	/**
 	 * {@inheritDoc ModelContainerRuntimeFactory.containerInitializingFirstTime}
 	 */
 	protected async containerInitializingFirstTime(runtime: IContainerRuntime): Promise<void> {
-		const taskList = await runtime.createDataStore(TaskListInstantiationFactory.type);
-		await taskList.trySetAlias(taskListId);
+		const taskListCollection = await runtime.createDataStore(
+			TaskListCollectionInstantiationFactory.type,
+		);
+		await taskListCollection.trySetAlias(taskListCollectionAlias);
 	}
 
 	/**
 	 * {@inheritDoc ModelContainerRuntimeFactory.containerHasInitialized}
 	 */
-	protected async containerHasInitialized(runtime: IContainerRuntime): Promise<void> {
-		runtime.on("signal", (message) => {
-			// TODO: Check the message type? clientId?  And route to the TaskList for interpretation?
-			// Interpretation of the message contents should probably live on the TaskList to encapsulate
-			// knowledge of the task-specific data.
-		});
-	}
+	protected async containerHasInitialized(runtime: IContainerRuntime): Promise<void> {}
```

Comment:
Yeah it's down in createModel because the container doesn't have access to the taskList at instantiation and the taskList owns the logic of how to act on receiving the signal. When I first wrote this, it felt icky back then too, but it's just the way it is right now.